src/pages/main/components/selected-dataset/build-flow/components/cards/description-card.tsx

```diff
@@ -29,6 +40,28 @@ export const DescriptionCard = (props: ICardProps) => {
   const history = useHistory()
   const ds = title || datasetStore.datasetName
 
+  const note = descriptions[ds] || dirinfoStore.dirInfoData?.dsDict[ds]?.note
+  const isOpenButton = optionsForOpenButton.includes(selectedValue)
+  const [isEditMode, setEditMode] = useState(false)
+  const [isTyping, setTyping] = useState(false)
+  const [datasetDescription, setDatasetDescription] = useState(
+    note || datasetDescriptionDefault,
+  )
+
+  const editDescriptionIcon = note?.length ? (
+    <Icon name="Edit" className={cn('text-blue-bright')} />
```

Comment:
Why you move JSX from return part, it seems that those conditional output will be readable there.